vectordraw.js

```diff
@@ -297,25 +394,34 @@ VectorDraw.prototype.getVectorSettingsByName = function(name) {
 
 VectorDraw.prototype.updateVectorProperties = function(vector) {
     var vec_settings = this.getVectorSettingsByName(vector.name);
-    var length_factor = vec_settings.length_factor || 1;
-    var length_units = vec_settings.length_units || '';
-    var base_angle = vec_settings.base_angle || 0;
     var x1 = vector.point1.X(),
         y1 = vector.point1.Y(),
         x2 = vector.point2.X(),
         y2 = vector.point2.Y();
-    var length = length_factor * Math.sqrt(Math.pow(x2-x1, 2) + Math.pow(y2-y1, 2));
-    var angle = ((Math.atan2(y2-y1, x2-x1)/Math.PI*180) - base_angle) % 360;
+    var length = vec_settings.length_factor * Math.sqrt(Math.pow(x2-x1, 2) + Math.pow(y2-y1, 2));
+    var slope = (y2-y1)/(x2-x1);
+    var angle = ((Math.atan2(y2-y1, x2-x1)/Math.PI*180) - vec_settings.base_angle) % 360;
     if (angle < 0) {
         angle += 360;
     }
     $('.vector-prop-name .value', this.element).html(vector.point2.name); // labels are stored as point2 names
-    $('.vector-prop-length .value', this.element).html(length.toFixed(2) + ' ' + length_units);
-    $('.vector-prop-angle .value', this.element).html(angle.toFixed(2));
+    if (vector.elType !== "line") {
+        $('.vector-prop-length', this.element).show();
+        $('.vector-prop-slope', this.element).hide();
+        $('.vector-prop-length .value', this.element).html(length.toFixed(2) + ' ' + vec_settings.length_units);
+        $('.vector-prop-angle .value', this.element).html(angle.toFixed(2));
+    }
+    else {
+        $('.vector-prop-length', this.element).hide();
```

Comment:
@smarnach angle is shown for lines, but not updated - so it's frozen to the last displayed value